packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2520,58 +2522,138 @@ function warnIfNotCurrentlyActingUpdatesInDEV(fiber: Fiber): void {
+            case SimpleMemoComponent:
+              if (
+                workInProgressNode.memoizedState !== null &&
+                workInProgressNode.memoizedState.baseUpdate !== null
+              ) {
+                let update = workInProgressNode.memoizedState.baseUpdate;
+                // Loop through the functional component's memoized state to see whether
+                // the component has triggered any high pri updates
+                while (update !== null) {
+                  const priority = update.priority;
+                  if (
+                    priority === UserBlockingPriority ||
+                    priority === ImmediatePriority
+                  ) {
+                    if (componentsThatTriggeredHighPriSuspend === null) {
+                      componentsThatTriggeredHighPriSuspend = new Set([
+                        getComponentName(workInProgressNode.type),
+                      ]);
+                    } else {
+                      componentsThatTriggeredHighPriSuspend.add(
+                        getComponentName(workInProgressNode.type),
+                      );
+                    }
+                    break;
+                  }
+                  if (
+                    update.next === workInProgressNode.memoizedState.baseUpdate
+                  ) {
+                    break;
+                  }
+                }
+              }
+              break;
+            default:
+              break;
+          }
+        }
+        workInProgressNode = workInProgressNode.return;
+      }
+
       // Add the component name to a set.
       const componentName = getComponentName(sourceFiber.type);
-      if (componentsWithSuspendedDiscreteUpdates === null) {
-        componentsWithSuspendedDiscreteUpdates = new Set([componentName]);
+      if (componentsThatSuspendedAtHighPri === null) {
+        componentsThatSuspendedAtHighPri = new Set([componentName]);
       } else {
-        componentsWithSuspendedDiscreteUpdates.add(componentName);
+        componentsThatSuspendedAtHighPri.add(componentName);
       }
     }
   }
 }
 
 function flushSuspensePriorityWarningInDEV() {
   if (__DEV__) {
-    if (componentsWithSuspendedDiscreteUpdates !== null) {
+    if (
+      componentsThatSuspendedAtHighPri !== null &&
+      componentsThatTriggeredHighPriSuspend !== null
+    ) {
       const componentNames = [];
-      componentsWithSuspendedDiscreteUpdates.forEach(name => {
+      componentsThatSuspendedAtHighPri.forEach(name => {
         componentNames.push(name);
       });
-      componentsWithSuspendedDiscreteUpdates = null;
+      componentsThatSuspendedAtHighPri = null;
+
+      const componentsThatTriggeredSuspendNames = [];
+      componentsThatTriggeredHighPriSuspend.forEach(name =>
+        componentsThatTriggeredSuspendNames.push(name),
+      );
+
+      const componentsThatTriggeredSuspendErrorMessage = componentsThatTriggeredHighPriSuspend.has(
+        HostRoot,
```

Comment:
How about we omit the root from this warning entirely? We can handle that separately inside `ReactDOM.render` and corresponding APIs (in a different PR), which will allow us to provide a more helpful warning for each renderer. This case is relatively rare, anyway.